tests/ast/data/DataFrame.stat.test

```diff
@@ -0,0 +1,91 @@
+## TEST CASE
+
+df = session.table("test_table")
+
+df2 = df.stat.approx_quantile("NUM", [0.5])
+
+df3 = df.stat.approx_quantile(["NUM", "NUM"], [0, 0.2, 0.4], statement_params={'a':'b'})
+
+df = session.create_dataframe([[0.1, 0.5], [0.2, 0.6], [0.3, 0.7]], schema=["a", "b"])
+
+df4 = df.stat.cov("a", "b")
+
+df5 = df.stat.cov("a", "b", statement_params={'a':'b'})
+
+df4 = df.stat.corr("a", "b")
+
+df5 = df.stat.corr("a", "b", statement_params={'a':'b'})
+
+df = session.create_dataframe([(1, 1), (1, 2), (2, 1), (2, 1), (2, 3), (3, 2), (3, 3)], schema=["key", "value"])
+
+ct = df.stat.crosstab("key", "value")
+
+ct2 = df.stat.crosstab("key", "value", statement_params={'a':'b'})
+
+df = session.create_dataframe([("Bob", 17), ("Alice", 10), ("Nico", 8), ("Bob", 12)], schema=["name", "age"])
+
+fractions = {"Bob": 0.5, "Nico": 1.0}
+
+sample_df = df.stat.sample_by("name", fractions)
+
+## EXPECTED OUTPUT
+
+df = session.table("test_table")
+
+df2 = df.stat
```

Comment:
Let's do the `na` approach here and not bind `stat`. This stat object is stateless, like na, unlike write.

Reply:
I already merged the server-side branch. We certainly can do the `na` approach here, however imho the more correct one should be similar to `write` as you can assign the `DataframeStatFunctions` object technically. If your concern is formatting, I believe your work with the inliner will resolve this. I can change everything similar to `na`, yet this means I need to reopen a server-branch, refactor, modify IR and modify this branch here. I estimate the work to be `~ 3h`. Do you think atm this a time investment worth it? If so sure - but I think we can always refactor this later if need be. Another option is open up a low priority JIRA ticket and do this refactor in another branch pair.

Reply:
This unnecessarily complicates the AST, making our Phase 2 work a lot more complicated than it should be. This is a super simple fix, but feel free to merge as-is.